domain/src/main/kotlin/no/nav/su/se/bakover/domain/visitor/LagBrevRequestVisitor.kt

```diff
@@ -325,17 +326,26 @@ class LagBrevRequestVisitor(
     }
 
     override fun visit(revurdering: AvsluttetRevurdering) {
-        brevRequest = hentPersonOgNavn(
-            // siden avslutt-brevet er et informasjons-brev, trengs ikke attestant
-            fnr = revurdering.fnr, saksbehandler = revurdering.saksbehandler, attestant = null,
-        ).map {
-            LagBrevRequest.AvsluttRevurdering(
-                person = it.person,
-                fritekst = revurdering.fritekst,
-                saksbehandlerNavn = it.saksbehandlerNavn,
-                dagensDato = LocalDate.now(clock),
-                saksnummer = revurdering.saksnummer,
-            )
+        val fritekst = when (revurdering.brevvalg) {
+            is Brevvalg.SkalIkkeSendeBrev, is Brevvalg.SaksbehandlersValg.SkalIkkeSendeBrev -> null
+            is Brevvalg.SkalSendeBrev -> revurdering.brevvalg.fritekst
+            is Brevvalg.SaksbehandlersValg.SkalSendeBrev.MedFritekst -> revurdering.brevvalg.fritekst
+        }
+        if (fritekst == null) {
```

Comment:
Noe som kan gjøres med apiet for å unngå at vi både må til med `when` og `if` for denne typen ting?

Reply:
Her manglet det en null uansett. Jeg endte opp med å kalle `brevvalg.skalSendesBrev()` og hvis ikke kaste exception.